src/openjd/model/_internal/_create_job.py

```diff
@@ -13,6 +13,25 @@
 
 __all__ = ("instantiate_model",)
 
+# Cache for TypeAdapter instances
+_type_adapter_cache: Dict[str, TypeAdapter] = {}
+
+
+def get_type_adapter(field_type: Any) -> TypeAdapter:
+    """Get a TypeAdapter for the given field type, using a cache for efficiency.
+
+    Args:
+        field_type: The type to adapt.
+
+    Returns:
+        A TypeAdapter for the given type.
+    """
+    # Use the string representation of the type as a cache key
+    type_key = str(field_type)
```

Comment:
Did you try `id` as an alternative to `str` as well? If types aren't too redundant, it might be a bit faster.